applications/asset_tracker/src/env_sensors/CMakeLists.txt

```diff
@@ -5,4 +5,31 @@
 #
 
 zephyr_include_directories(.)
-target_sources(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/env_sensors.c)
+
+if (CONFIG_USE_BME680_BSEC)
+	if(CONFIG_FP_HARDABI)
+		target_sources(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/bsec.c)
+		set (bsec_dir ${NRF_DIR}/ext/BSEC_1.4.7.4_Generic_Release)
+		set(BSEC_LIB_DIR     ${bsec_dir}/algo/Normal_version/bin/gcc/Cortex_M4F)
+		set(BSEC_INCLUDE_DIR ${bsec_dir}/algo/Normal_version/bin/gcc/Cortex_M4F)
+		if(NOT EXISTS "${BSEC_LIB_DIR}/libalgobsec.a")
+			assert(0 "Could not find BSEC library")
+		endif()
+		target_include_directories(app PRIVATE ${bsec_dir}/API)
+		target_include_directories(app PRIVATE ${bsec_dir}/examples)
+		target_include_directories(app PRIVATE BSEC_INCLUDE_DIR)
+		target_sources(app PRIVATE ${bsec_dir}/examples/bsec_integration.c)
+		target_sources(app PRIVATE ${bsec_dir}/API/bme680.c)
+
+		add_library(bsec_lib STATIC IMPORTED GLOBAL)
+		add_dependencies(bsec_lib math_lib bsec_target)
+		set_target_properties(bsec_lib PROPERTIES IMPORTED_LOCATION "${BSEC_LIB_DIR}/libalgobsec.a")
+		set_target_properties(bsec_lib PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${BSEC_INCLUDE_DIR})
+		target_link_libraries(bsec_lib INTERFACE -L${LIBC_LIBRARY_DIR})
+	else()
+		assert(0 "CONFIG_FB_HARDABI must be set")
+	endif()
```

Comment:
Are we / should we set FP_HARDABI from the same prj.conf that sets USE_BME680_BSEC ?

Reply:
I am not sure. As this is inside the Asset tracker application it will most likely be run with this enabled. I added this to give an understandable error message before it would fail during linking.